Pet-Clinic-Frontend/src/pages/Registration.tsx

```diff
@@ -1,49 +1,116 @@
-
 import { useState } from 'react';
-
-import './Registration.css'
+import './Registration.css';
 
 const Registration = () => {
+    const [email, setEmail] = useState("");
+    const [password, setPassword] = useState("");
+    const [error, setError] = useState<string | null>(null); //generic type <string | null> specify that the state managed by useState can be a string or null. Initial State (null)
+    const [submitted, setSubmitted] = useState(false);
+
+    const handleEmail = (e: React.ChangeEvent<HTMLInputElement>) => {
+        setEmail(e.target.value);
+        setSubmitted(false);
+    }
+
+    const handlePassword = (e: React.ChangeEvent<HTMLInputElement>) => {
+        setPassword(e.target.value);
+        setSubmitted(false);
+    }
+
+    const validateForm = () => {
+        if (!email || !password) {
+            //If email is an empty string, null, or undefined, !email will be true.
+            // When email is evaluated, if email is an empty string (""), null, or undefined, it is considered falsy. 
+            setError("Email and password are required");
+            return false;
+        }
+        if (!/\S+@\S+\.\S+/.test(email)) {  //syntax /\S+@\S+\.\S+/ is a regular expression (regex) https://regexr.com //.test(email): This method tests whether the email string matches the regular expression pattern. 
```

Comment:
I am not the biggest fan of the comments here. If you make variables and functions with the "good" names, you will avoid duplicating things.

Reply:
Comments deleted. isValidEmail function with proper naming variables was added.